src/tools/interop/idt/capture/ecosystem/play_services/play_services.py

```diff
@@ -87,10 +98,23 @@ async def start_capture(self) -> None:
             verbose_command = f"shell setprop log.tag.gms_svc_id:{service_id} VERBOSE"
             self.platform.run_adb_command(verbose_command)
         self._get_standard_info()
-        await self.platform.start_streaming()
+        if config.test_error_execution:
+            self.logger.critical("Throwing exception in execution for test!")
+            raise Exception("Test exe exception from Play Services!")
 
     async def stop_capture(self) -> None:
-        await self.platform.stop_streaming()
-
-    async def analyze_capture(self) -> None:
-        self.analysis.do_analysis()
+        self.analysis.show_analysis()
+        if self.logcat_fd is not None:
+            self.logcat_fd.close()
+
+    async def analyze_capture(self):
+        self.logcat_fd = open(self.platform.streams["LogcatStreamer"].logcat_artifact, "r")
+        while True:
+            self.analysis.do_analysis(self.logcat_fd.readlines())
+            await asyncio.sleep(0)
```

Comment:
Is this an indication of some weird non-asyc stuff that you're forcing or something there? Can you add a comment about what this is doing?

Reply:
Yes I'm releasing the event loop. https://docs.python.org/3/library/asyncio-task.html#sleeping

Other options include adding a dependency that provides awaitable readlines or implementing one

Reply:
Added this comment to both places this pattern exists!